compose/material3/material3/src/skikoMain/kotlin/androidx/compose/material3/Strings.skiko.kt

```diff
@@ -96,9 +96,9 @@ internal actual fun getString(string: Strings): String {
 // TODO current implementation doesn't support sophisticated formatting like %.2f,
 //  but currently we use it only for integers and strings
 internal actual fun String.format(vararg formatArgs: Any?): String {
-    val result = this
+    var result = this
     formatArgs.forEachIndexed { index, arg ->
-        result.replace("%$index\$", arg.toString())
+        result = result.replace("%${index+1}$", arg.toString())
```

Comment:
There were some bugs in `String.format` implementation:

- `String.replace` does not mutate string. It creates new one 
- Args counting in string starts from 1 (not from 0):
- https://github.com/JetBrains/compose-multiplatform-core/blob/f516176c383760c9b445d22f7f91098496020bbd/compose/material3/material3/src/skikoMain/kotlin/androidx/compose/material3/Strings.skiko.kt#L60

Before: 
<img width="380" alt="Screenshot 2023-08-01 at 14 06 58" src="https://github.com/JetBrains/compose-multiplatform-core/assets/63979218/44333387-c587-4ddc-85a8-7172c0ecd51d">

After: 

<img width="378" alt="Screenshot 2023-08-01 at 14 13 13" src="https://github.com/JetBrains/compose-multiplatform-core/assets/63979218/7ae3c248-b8a8-4c68-82cc-b1d2e5ce8f40">